tests/test_cli.py

```diff
@@ -330,6 +330,19 @@ def test_keep_stdout(self):
         self.run_decompress_stdout([str(self.compressed_path), "-c"])
         self.assertTrue(self.compressed_path.exists())
 
+    def test_valid_stdout(self):
+        stdout = sys.__stdout__
+        try:
+            tmp_file = pathlib.Path(self.tmpdir.name) / "stdout.trees"
+            sys.__stdout__ = open(tmp_file, "w")
+            cli.tszip_main(["-d", "-c", str(self.compressed_path)])
+            ts = tskit.load(str(tmp_file))
+            assert ts.tables == self.ts.tables
+            self.assertTrue(self.compressed_path.exists())
+        finally:
+            sys.__stdout__.close()
+            sys.__stdout__ = stdout
+
```

Comment:
@jeromekelleher I had to implement something like this to avoid capturing conflicts with `pytest`, but it does work at least on my Macbook Air. Would be interested in thoughts to make this a more general function to capture and/or redirect stdout within pytest.